client/core/bookie.go

```diff
@@ -1076,6 +1202,16 @@ func handleEpochOrderMsg(_ *Core, dc *dexConnection, msg *msgjson.Message) error
 		return fmt.Errorf("epoch order note unmarshal error: %w", err)
 	}
 
+	// Acquiring dc.booksMtx here in dc.bookie serves as a barrier for notifications that
+	// come after we sent new subscription request (e.i. notes that need to be applied
+	// on top of snapshot we get from new subscription request will wait on that mutex
+	// here). Note, we don't acquire dc.booksMtx for the whole handleBookOrderMsg
+	// duration; that leaves some room for notifications from previous subscription to
+	// come and apply concurrently with us applying snapshot we just got from new
+	// subscription request.
+	// TODO: I'm not sure at the moment whether is OK to execute this notification from
+	//  old subscription while Resetting order book concurrently. Could this result into
+	//  mixed OrderBook state (for orders, candles cache, or something else) ?
```

Comment:
Would like to know answer to that TODO question (that's in `handleEpochOrderMsg ` **func**).

Reply:
It does seem to trigger this error:
```go
ob.log.Errorf("epoch order note received for epoch %d but current epoch is %d", idx, ob.currentEpoch)
```
because new order book snapshot might have higher `currentEpoch` value when outdated `epoch_order` applies,

so if applying outdated `epoch_order` is the expected behavior, it probably should be logged as error

Reply:
Epoch orders don't seem to be interfering with outstanding orders, so looks like `epoch_order` notification is meant to be executing concurrently with the reset of order book updates (which it currently is).

Reply:
On closer inspection, it seems epoch orders (`epoch_orders` notification and other related stuff such as `epochQueues`, `currentEpoch` and `proofedEpoch`) are kinda half-baked now:
- it doesn't have anything to do with `orders` or `seq`, but for some reason `seq` is incremented with each `epoch_orders` notification; we should probably remove `seq` usage by `epoch_orders` notification
- but similar to `orders`<->`seq` [atomic relationship](https://github.com/norwnd/dcrdex/pull/2) `epochQueues`, `currentEpoch` and `proofedEpoch` could benefit from atomicity because currently these fall  out of sync quite often, which results in spamming something like this in logs (and what's more important long term, does not allow client to differentiate between server sending invalid match proof note and client's view on epoch orders being out of sync = non-atomic):
```
2023-03-09 05:28:53.450 [ERR] CORE: Route 'match_proof' notification handler error (DEX dex.decred.org:7232): match proof validation failed: epoch 167833983 match proof note references 1 orders, but local epoch queue is empty
...
2023-03-09 14:41:50.809 [ERR] CORE: Route 'match_proof' notification handler error (DEX dex.decred.org:7232): match proof validation failed: epoch 167837301 match proof note references 3 orders, but local epoch queue has 2
```
that atomicity isn't trivial to implement though ^ (and whether it should be done with current `subscribe/resubscribe` mechanisms or have a completely separate parallel machinery working is something to investigate) and it's not high priority any time soon; so I'm downgrading log level to `Trace` in this PR and supply relevant comments there too since it isn't useful to anyone to see as error the way it is currently implemented.

Reply:
By the way, saw relevant discussions somewhere around https://github.com/decred/dcrdex/pull/278#pullrequestreview-401038421, but nothing conclusive.